sic_io/src/save.rs

```diff
@@ -1,34 +1,36 @@
+use std::io::Write;
 use std::path::Path;
 
 use sic_core::image;
 
 use crate::conversion::{AutomaticColorTypeAdjustment, ConversionWriter};
 
-pub fn export<P: AsRef<Path>>(
+pub fn export<W: Write>(
     image: &image::DynamicImage,
-    method: ExportMethod<P>,
+    // method: ExportMethod<P>,
+    writer: &mut W,
     format: image::ImageOutputFormat,
     export_settings: ExportSettings,
 ) -> Result<(), String> {
-    let writer = ConversionWriter::new(image);
-    writer.write(method, format, export_settings.adjust_color_type)
+    let conv = ConversionWriter::new(image);
+    conv.write(writer, format, export_settings.adjust_color_type)
 }
 
 #[derive(Debug)]
 pub struct ExportSettings {
     pub adjust_color_type: AutomaticColorTypeAdjustment,
 }
 
-#[derive(Debug)]
-pub enum ExportMethod<P: AsRef<Path>> {
-    File(P),
-    StdoutBytes,
-}
+// #[derive(Debug)]
+// pub enum ExportMethod<P: AsRef<Path>> {
+//     File(P),
+//     StdoutBytes,
+// }
 
-// enum variants on type aliases are currently experimental, so we use a function here instead.
-pub fn use_stdout_bytes_as_export_method() -> ExportMethod<EmptyPath> {
-    ExportMethod::StdoutBytes
-}
+// // enum variants on type aliases are currently experimental, so we use a function here instead.
```

Comment:
Commented original code (18)

(I marked them all to have a quick overview to see them all marked as changed =); I'll try to do a proper review by tomorrow)